pkg/util/file_indexer.go

```diff
@@ -302,3 +307,311 @@ func WriteFile(newFileMap map[string]FileData, resolvedPath string) error {
 
 	return err
 }
+
+func RunIndexWithRemote(directory string, ignoreRules []string, remoteDirectories map[string]string) (ret IndexerRet, err error) {
```

Comment:
I see common things between RunIndex and this function, can we consider writing a nice struct and follow a design pattern? maybe extract some functions from this..?

Reply:
I can suggest a structure, but tomorrow. Need to understand a bit of this code 🙂

Reply:
Current the PR is `WIP` and I am trying out certain things. Once ready, I will probably remove `RunIndex`.

Reply:
Are you planning to remove the RunIndex function? because still there seem to be common code between the functions?

Reply:
No, I am currently not planning to remove the RunIndex function. I was planning to leave it untouched for s2i components.

Reply:
/hold cancel